EIPS/eip-4337.md

```diff
@@ -0,0 +1,313 @@
+---
+eip: 4337
+title: Account Abstraction via Entry Point Contract specification
+author: Vitalik Buterin, Yoav Weiss, Kristof Gazso, Namra Patel, Dror Tirosh
+discussions-to: https://ethereum-magicians.org/t/erc-4337-account-abstraction-via-entry-point-contract-specification/7160
+type: Standards Track
+category: ERC
+status: Draft
+created: 2021-09-29
+---
+
+## Simple Summary
+
+An account abstraction proposal which completely avoids the need for consensus-layer protocol changes, instead relying on a separate mempool of `UserOperation` objects and miners either running custom code or connecting to a bundle marketplace.
+
+## Motivation
+
+See also: https://ethereum-magicians.org/t/implementing-account-abstraction-as-part-of-eth1-x/4020 and the links therein for historical work and motivation, and [EIP 2938](https://eips.ethereum.org/EIPS/eip-2938) for a consensus layer proposal for implementing the same goal.
+
+This proposal takes a different approach, avoiding any adjustments to the consensus layer. It seeks to achieve the following goals:
+
+* **Achieve the key goal of account abstraction**: allow users to use smart contract wallets containing arbitrary verification logic instead of EOAs as their primary account. Completely remove any need at all for users to also have EOAs (as status quo SC wallets and EIP 3074 both require)
```

Comment:
```suggestion
* **Achieve the key goal of account abstraction**: allow users to use smart contract wallets containing arbitrary verification logic instead of EOAs as their primary account. Completely remove any need at all for users to also have EOAs (as status quo SC wallets and EIP-3074 both require)
```